### DISCUSSION TITLE
Order should have link the Cancellation event

### DISCUSSION THREAD

**Discussion by BLR-0118** [beckn/protocol-specifications#281] [Core Specification]
Currently, the Order is not linked to the Cancellation event, due to which attributes such as cancellation_reason_id and cancelled_by, are not available to the BAP/BPP that are a party to the Order.
Order should be linked to the Cancellation event such that cancellation related info is available to BAP/BPP that are a party to the Order.

**Comment by ravi-prakash-v** [comment]
@BLR-0118 the cancellation event can be linked to the order **_view_** on the BAP's UI. 

The `Order` object stores the current state of the order. It does not store the event history of the order. It is the BAP and the BPP that must persist the event history on their respective ends. The `order.state` currently defines if the order is cancelled, active or complete. 

The rendering of an order may be unified but IMO, attaching the event with the order may be a bit of an overkill at the protocol layer. What do you think?

**Comment by ravi-prakash-v** [comment]
Can you elaborate on the use cases that will involve these attributes?
